kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/SchemaUtils.java

```diff
@@ -52,9 +54,26 @@ private SchemaUtils() {}
    * @throws IllegalArgumentException if the schema is invalid
    */
   public static void validateSchema(StructType schema, boolean isColumnMappingEnabled) {
+    Set<TableFeature> features =
+        isColumnMappingEnabled
+            ? new HashSet<>(Collections.singletonList(TableFeatures.COLUMN_MAPPING_RW_FEATURE))
+            : new HashSet<>();
+    validateSchema(schema, features);
+  }
+
+  /**
+   * Validate the schema against the set of provided features. This method checks if the schema has
+   * no duplicate columns, the names contain only valid characters and the data types are supported.
+   *
+   * @param schema the schema to validate
+   * @param features the table features to validate the schema against
+   * @throws IllegalArgumentException if the schema is invalid
+   */
+  public static void validateSchema(StructType schema, Set<TableFeature> features) {
     checkArgument(schema.length() > 0, "Schema should contain at least one column");
-    validateColumnNames(schema, isColumnMappingEnabled);
-    validateSupportedType(schema);
+    checkArgument(features != null, "Table features should not be null");
+    validateColumnNames(schema, features.contains(TableFeatures.COLUMN_MAPPING_RW_FEATURE));
```

Comment:
The COLUMN_MAPPING_RW_FEATURE doesn't necessarily imply column mapping is enabled in theory, this value should really come from the metadata configuration.